compiler/src/dotty/tools/dotc/core/TypeErasure.scala

```diff
@@ -687,6 +689,10 @@ class TypeErasure(sourceLanguage: SourceLanguage, semiEraseVCs: Boolean, isConst
       tp
   }
 
+  private def underlyingOfTermRef(tp: TermRef)(using Context) = tp.widen match
```

Comment:
This probably deserves a comment referencing the issue. Otherwise someone will come here later, figure that this is useless, and remove it.